apps/aecore/src/aec_vm_chain.erl

```diff
@@ -468,20 +507,72 @@ do_set_store(Store, PubKey, Trees) ->
         end,
     aect_state_tree:enter_contract(NewContract, ContractsTree).
 
-apply_transaction(Tx, #state{ trees = Trees, tx_env = Env } = State) ->
+apply_transaction(Tx, #state{tx_env = Env } = State) ->
+    Trees = get_top_trees(State),
```

Comment:
So execution of state-modifying callbacks (spend, oracle register/query/respond/extend, aens preclaim/claim/transfer/revoke) in force progress tx (contract call update) read&write only off chain trees, right? It means it will not [throw exception](https://github.com/aeternity/protocol/pull/215/files#r221468884)?

Reply:
It is to be checked if those can be used for removing tokens from other account - that then get in channel state on chain - though that applies to all contract calls anyway.

Reply:
Modifying of trees that are not expected to be modified is tracked as PT [SC contract: register a name (negative test)](https://www.pivotaltracker.com/story/show/160992836)